processor/processor.go

```diff
@@ -0,0 +1,83 @@
+package processor
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/gonative-cc/relayer/dal"
+	"github.com/gonative-cc/relayer/ika"
+)
+
+// NativeTxData represents the data extracted from a Native chain transaction.
+type NativeTxData struct {
+	TxID           uint64   `json:"tx_id"`
+	DWalletCapID   string   `json:"dwallet_cap_id"`
+	SignMessagesID string   `json:"sign_messages_id"`
+	Messages       [][]byte `json:"messages"`
+}
+
+// Processor handles processing transactions from the Native chain to IKA for signing.
+type Processor struct {
+	ikaClient ika.IClient
+	db        *dal.DB
+}
+
+// NewProcessor creates a new Processor instance.
+func NewProcessor(ikaClient ika.IClient, db *dal.DB) *Processor {
```

Comment:
Why is ikaClient not a pointer? I'm curious; not a suggestion.

Reply:
Because its an interface

Reply:
Ah, yes!